clients/src/test/java/org/apache/kafka/clients/producer/KafkaProducerTest.java

```diff
@@ -952,7 +953,8 @@ public void testTopicRefreshInMetadata() throws InterruptedException {
                 }
             });
             t.start();
-            assertThrows(TimeoutException.class, () -> producer.partitionsFor(topic));
+            Throwable throwable = assertThrows(TimeoutException.class, () -> producer.partitionsFor(topic));
+            assertEquals(UnknownTopicOrPartitionException.class, throwable.getCause().getClass());
```

Comment:
what about `assertInstanceOf(UnknownTopicOrPartitionException.class, throwable.getCause());`. Seems clearer and safer (test would fail instead of throwing NPE if cause ends up being null by mistake)